Makefile

```
@@ -82,6 +82,15 @@ test-failpoint:
 	BBOLT_VERIFY=all TEST_FREELIST_TYPE=array go test -v ${TESTFLAGS} -timeout 30m ./tests/failpoint
 
 .PHONY: test-robustness # Running robustness tests requires root permission
-test-robustness:
+test-robustness: check-gofail
 	go test -v ${TESTFLAGS} ./tests/dmflakey -test.root
 	go test -v ${TESTFLAGS} ${ROBUSTNESS_TESTFLAGS} ./tests/robustness -test.root
+
+.PHONY: check-gofail
+check-gofail: # Returns fail if binaries aren't built with gofail
+	@(for binary_name in bbolt; do \
+		export BINARY_PATH=`command -v $$binary_name`; \
+		go version -m $$BINARY_PATH | grep -q go.etcd.io/gofail && continue; \
+		sh -c "echo $$BINARY_PATH wasn\'t built with gofail 1>&2"; \
```

Suggest we be explicit about what the contributor needs to do to remove any guesswork.

```suggestion
		sh -c "echo $$BINARY_PATH wasn\'t built with gofail, run make gofail-enable first 1>&2"; \
```

---

To be clearer, we only need to execute `make gofail-enable` when building the bbolt binary instead of before running the test case. PLease let's clearly clarify it in the log.